ui/components/DagGraphNode.tsx

```diff
@@ -0,0 +1,127 @@
+import { Tooltip } from "@material-ui/core";
+import * as React from "react";
+import styled from "styled-components";
+import images from "../lib/images";
+import { FluxObjectNode } from "../lib/objects";
+import Flex from "./Flex";
+import { computeReady, ReadyType } from "./KubeStatusIndicator";
+
+type Props = {
+  className?: string;
+  object?: FluxObjectNode;
+  isCurrentNode?: boolean;
+};
+
+const nodeBorderRadius = 30;
+const titleFontSize = "48px";
+const kindFontSize = "36px";
+
+const GraphIcon = styled.img`
+  height: ${titleFontSize};
+  width: ${titleFontSize};
+  min-height: ${titleFontSize};
+  min-width: ${titleFontSize};
+`;
+
+const Node = styled(Flex)`
+  background: white;
+  border: 5px solid ${(props) => props.theme.colors.neutral30};
+  border-radius: ${nodeBorderRadius}px;
+  user-select: none;
+`;
+
+const NodeText = styled(Flex)`
+  width: 90%;
+  align-items: flex-start;
+  justify-content: space-evenly;
+`;
+
+const Title = styled(Flex)`
+  font-size: ${titleFontSize};
+`;
+
+const Kinds = styled(Flex)`
+  font-size: ${kindFontSize};
+  color: ${(props) => props.theme.colors.neutral30};
+`;
+
+type StatusLineProps = {
+  suspended: boolean;
+  status: ReadyType;
+};
+
+const StatusLine = styled.div<StatusLineProps>`
+  width: 5%;
+  height: 100%;
+  border-radius: ${nodeBorderRadius - 4.5}px 0 0 ${nodeBorderRadius - 4.5}px;
+  background-color: ${(props) => {
+    if (props.suspended) return props.theme.colors.suspended;
+    else if (props.status === ReadyType.Ready)
+      return props.theme.colors.success;
+    else if (props.status === ReadyType.Reconciling)
+      return props.theme.colors.primary10;
+    else if (props.status === ReadyType.NotReady)
+      return props.theme.colors.alert;
+    else return "transparent";
+  }};
+`;
+
+function getStatusIcon(status: ReadyType, suspended: boolean) {
+  if (suspended) return <GraphIcon src={images.suspendedSrc} />;
+  switch (status) {
+    case ReadyType.Ready:
+      return <GraphIcon src={images.successSrc} />;
+
+    case ReadyType.Reconciling:
+      return <GraphIcon src={images.reconcileSrc} />;
+
+    case ReadyType.NotReady:
+      return <GraphIcon src={images.failedSrc} />;
+
+    default:
+      return "";
+  }
+}
+
+const StyledObjectName = styled.span<{ isCurrentNode: boolean }>`
```

Comment:
> is there a chance

There's always a chance :grin:

Though, now that I've seen it, I kind-of want this feature in the regular graph as well - did we not have something like it once?

> Passed object kind has also changed to FluxObjectNode

Ugh, so it _would_ work, but it would kinda only work by accident as the types are completely independent? That's unfortunate - perhaps that should have further refactoring done to unify the the types before we slam them into the same function.

Reply:
The old graph itself uses `descendants: any[];` and children used to create `descendants` are of type UnstructuredObject | any[]. Type of rootNode is... I didn't quite get it, there are not many type deifinitions.

The new graph uses FluxObejctNode[] as nodes prop, what is easier.

I could try to combine the graphs or more like replace the old graph with the new one in this PR, but there is a risk that the Dependencies will not make it into this release.

Reply:
@ozamosi 

>Though, now that I've seen it, I kind-of want this feature in the regular graph as well - did we not have something like it once?

Do you mean the arrows as "this feature" or anything else you want to have in the regular graph too?

Reply:
> Do you mean the arrows as "this feature" or anything else you want to have in the regular graph too?

Nono, I mean the `isCurrentNode` - which is the thing that changed from GraphNode?

> I could try to combine the graphs or more like replace the old graph with the new one in this PR

It sounds like you're saying they're false friends - they look almost the same, but they're not the same, but they could be made to be the same, but we ought to make the old code look more like the new code. So that sounds like it's out of scope for here.

Reply:
@ozamosi 

isCurrentNode is very useful for the new graph because the current not is not necessarily the root node so we can see it on any place in the graph and need to be able to tell immediately where the current node is.

But in the reconciliation graph the current node is located as the "second node" under the root node (which is the source). So, you would still like to mark the "second node" as the current node and display its name in bold?

Reply:
> It sounds like you're saying they're false friends - they look almost the same, but they're not the same, but they could be made to be the same, but we ought to make the old code look more like the new code.

Smth. like that, yeah. 🤣  OK, I will save it for after the next two releases then.

Reply:
For now combines both node components into `GraphNode`, this was much easier to do than combining the graphs. Now it supports both node data object formats and `isCurrentNode` field for both graphs.